versiondb/client/dump.go

```diff
@@ -86,8 +90,13 @@ func DumpChangeSetCmd(opts Options) *cobra.Command {
 				if err != nil {
 					return err
 				}
-				endVersion = latestVersion + 1
-				fmt.Println("end version not specified, default to latest version + 1,", endVersion)
+				if noSdk46Compact {
+					endVersion = latestVersion
+					fmt.Println("end version not specified, default to latest version,", endVersion)
+				} else {
+					endVersion = latestVersion + 1
+					fmt.Println("end version not specified, default to latest version + 1,", endVersion)
```

Comment:
can you explain the root cause for this?

Reply:
I didn't realize mem store [issue](https://github.com/cosmos/cosmos-sdk/issues/14916),  could close this pr now